src/main/java/ru/itmo/roguelike/characters/Actor.java

```diff
@@ -129,7 +126,11 @@ public void strike(int damage) {
         if (damage > 0) {
             new Splash(position, 3, drawableDescriptor.getColor());
         }
-        this.hp -= this.def * this.armorResistance * damage;
+        if (armor != null) {
+            this.hp -= this.def * damage * armor.getArmorResistance();
+        } else {
+            this.hp -= this.def * damage;
+        }
```

Comment:
It will be more convenient if you have method:
```java
public float Actor.getArmorResistance() {
  if (armor == null) return 1;

  return armor.getResistance();
}
```

Then you'd use it here like `this.hp -= this.def * damage * getArmorResistance()`

Reply:
Я не уверена, что этот метод нужен, если он будет использоваться только в одном месте